docs/getting-started/libnetwork/Demonstration.md

```diff
@@ -34,40 +36,50 @@ You should see output like this on each node
 
 ## Creating networked endpoints
 
-The experimental channel version of Docker introduces a new flag to 
-`docker run` to network containers:  `--publish-service <service>.<network>.<driver>`.
+As of Docker 1.9, the integration of Docker with libnetwork introduces a new
+mechanism to provide networking for Docker containers.
 
- * `<service>` is the name by which you want the container to be known on the network.
- * `<network>` is the name of the network to join.  Containers on different networks cannot communicate with each other.
- * `<driver>` is the name of the network driver to use.  Calico's driver is called `calico`.
+The new command `docker network` can be used to create a logical network.  
+A new flag is introduced to `docker run` to join a container to a particular 
+network:  `--net <network>`.
 
-So let's go ahead and start a few containers on each host.
+So let's go ahead and create some networks and start a few containers 
+on each host spread between these networks.
+
+On either host, create three networks:
+
+    docker network create --driver=calico net1
+    docker network create --driver=calico net2
+    docker network create --driver=calico net3
+    
+Note that we use the Calico driver `calico`.  This driver is run within 
+the calico-node container.
 
 On calico-01
 
-    docker run --publish-service srvA.net1.calico --name workload-A -tid busybox
-    docker run --publish-service srvB.net2.calico --name workload-B -tid busybox
-    docker run --publish-service srvC.net1.calico --name workload-C -tid busybox
+    docker run --net net1 --name workload-A -tid busybox
+    docker run --net net2 --name workload-B -tid busybox
+    docker run --net net1 --name workload-C -tid busybox
 
 On calico-02
 
-    docker run --publish-service srvD.net3.calico --name workload-D -tid busybox
-    docker run --publish-service srvE.net1.calico --name workload-E -tid busybox
+    docker run --net net3 --name workload-D -tid busybox
+    docker run --net net1 --name workload-E -tid busybox
 
 By default, networks are configured so that their members can communicate with 
 one another, but workloads in other networks cannot reach them.  A, C and E are
- all in the same network so should be able to ping each other.  B and D are in 
- their own networks so shouldn't be able to ping anyone else.
+all in the same network so should be able to ping each other.  B and D are in 
+their own networks so shouldn't be able to ping anyone else.
     
 On calico-01 check that A can ping C and E.
 
-    docker exec workload-A ping -c 4 srvC
-    docker exec workload-A ping -c 4 srvE
+    docker exec workload-A ping -c 4 workload-C
```

Comment:
I don't think hostname lookups will work for containers in different networks. Maybe you can fully qulify them (workload-d.net3) Look in /etc/hosts